app/controllers/actions/PspDataRetrievalAction.scala

```diff
@@ -37,16 +38,17 @@ class PspDataRetrievalImpl @Inject()(val viewConnector: SchemeDetailsReadOnlyCac
   override protected def transform[A](request: AuthenticatedRequest[A]): Future[OptionalDataRequest[A]] = {
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
+    def isSchemeDataForCurrentSRN(data: JsValue): Boolean = (data \ SchemeSrnId.toString).asOpt[String].contains(srn)
+
     viewConnector.fetch(request.externalId).flatMap {
-      case None =>
+      case Some(data) if isSchemeDataForCurrentSRN(data) =>
+        Future.successful(OptionalDataRequest(
+          request.request, request.externalId, Some(UserAnswers(data)), request.psaId, request.pspId, viewOnly = true, request.administratorOrPractitioner))
+      case _ =>
         val pspId = request.pspId.getOrElse(throw IdNotFound("PspIdNotFound")).id
         schemeDetailsConnector.getPspSchemeDetails(pspId, srn).map { ua =>
           OptionalDataRequest(request.request, request.externalId, Some(additionalUserAnswers(ua)), request.psaId, request.pspId, viewOnly = true, request.administratorOrPractitioner)
```

Comment:
def isSchemeDataForCurrentSRN was created but never called?

Reply:
It is called, just below

Reply:
Sorry. Missed it.